src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptor.cs

```diff
@@ -14,139 +14,43 @@ namespace Microsoft.AspNet.Razor.TagHelpers
-                tagStructure: TagStructure.Unspecified,
-                designTimeDescriptor: null)
-        {
-        }
-
-        /// <summary>
-        /// Instantiates a new instance of the <see cref="TagHelperDescriptor"/> class with the given
-        /// <paramref name="attributes"/>.
-        /// </summary>
-        /// <param name="prefix">
-        /// Text used as a required prefix when matching HTML start and end tags in the Razor source to available
-        /// tag helpers.
-        /// </param>
-        /// <param name="tagName">The tag name that the tag helper targets. '*' indicates a catch-all
-        /// <see cref="TagHelperDescriptor"/> which applies to every HTML tag.</param>
-        /// <param name="typeName">The full name of the tag helper class.</param>
-        /// <param name="assemblyName">The name of the assembly containing the tag helper class.</param>
-        /// <param name="attributes">
-        /// The <see cref="TagHelperAttributeDescriptor"/>s to request from the HTML tag.
-        /// </param>
-        /// <param name="requiredAttributes">
-        /// The attribute names required for the tag helper to target the HTML tag.
-        /// </param>
-        /// <param name="allowedChildren">HTML element tag names targeted by tag helpers allowed as children.</param>
-        /// <param name="tagStructure">The expected tag structure.</param>
-        /// <param name="designTimeDescriptor">The <see cref="TagHelperDesignTimeDescriptor"/> that contains design
-        /// time information about the tag helper.</param>
-        public TagHelperDescriptor(
-            string prefix,
-            [NotNull] string tagName,
-            [NotNull] string typeName,
-            [NotNull] string assemblyName,
-            [NotNull] IEnumerable<TagHelperAttributeDescriptor> attributes,
-            [NotNull] IEnumerable<string> requiredAttributes,
-            IEnumerable<string> allowedChildren,
-            TagStructure tagStructure,
-            TagHelperDesignTimeDescriptor designTimeDescriptor)
-        {
-            Prefix = prefix ?? string.Empty;
-            TagName = tagName;
-            FullTagName = Prefix + TagName;
-            TypeName = typeName;
-            AssemblyName = assemblyName;
-            Attributes = new List<TagHelperAttributeDescriptor>(attributes);
-            RequiredAttributes = new List<string>(requiredAttributes);
-            TagStructure = tagStructure;
-            DesignTimeDescriptor = designTimeDescriptor;
-
-            if (allowedChildren != null)
-            {
-                AllowedChildren = new List<string>(allowedChildren);
-            }
-        }
-
-        /// <summary>
         /// Text used as a required prefix when matching HTML start and end tags in the Razor source to available
         /// tag helpers.
         /// </summary>
-        public string Prefix { get; }
+        public string Prefix { get; set; } = string.Empty;
```

Comment:
Disagree on this one @NTaylorMullen because the `TagHelperDescriptorResolver` sets this to `null` in the normal case. We want this to be `string.Empty` in that case.